ramani-maplibre/src/main/java/org/ramani/compose/Symbol.kt

```diff
@@ -55,19 +67,47 @@ fun Symbol(
                 .withIconImage(imageId.toString())
                 .withIconColor(color)
                 .withIconSize(size)
+                .withIconAnchor(imageAnchor)
                 .withIconRotate(imageRotation)
+                .withIconOffset(imageOffset)
         }
 
         text?.let {
             symbolOptions = symbolOptions
                 .withTextField(text)
-                .withTextColor(color)
+                .withTextColor(textColor)
+                .withTextHaloColor(textHaloColor)
+                .withTextHaloWidth(textHaloWidth)
                 .withTextSize(size)
-                .withTextJustify(TEXT_JUSTIFY_CENTER)
-                .withTextAnchor(TEXT_ANCHOR_CENTER)
+                .withTextJustify(textJustify)
+                .withTextAnchor(textAnchor)
+                .withTextOffset(textOffset)
         }
 
         val symbol = symbolManager.create(symbolOptions)
+
+        if (isDraggable) {
+            symbolManager.addDragListener(
+                object: OnSymbolDragListener {
+                    override fun onAnnotationDragStarted(annotation: Symbol?) {
+
+                    }
```

Comment:
Nitpick: can we keep this on one line?

```suggestion
                    override fun onAnnotationDragStarted(annotation: Symbol?) {}
```